catalog/rest/catalog-rest-endpoint/src/main/java/org/codice/ddf/endpoints/rest/RESTEndpoint.java

```diff
@@ -836,9 +858,128 @@ public Response addDocument(@Context HttpHeaders headers, @Context UriInfo reque
         return response;
     }
 
+    CreateInfo parseAttachments(List<Attachment> contentParts, String transformerParam) {
+
+        if (contentParts.size() == 1) {
+            Attachment contentPart = contentParts.get(0);
+            return parseAttachment(contentPart);
+        } else {
+            List<Attribute> attributes = new ArrayList<>(contentParts.size());
+            Metacard metacard = null;
+            CreateInfo createInfo = null;
+
+            for (Attachment attachment : contentParts) {
+                String name = attachment.getContentDisposition()
+                        .getParameter("name");
+
+                String parsedName = (name.startsWith("parse.")) ? name.substring(6) : name;
+                try {
+                    InputStream inputStream = attachment.getDataHandler()
```

Comment:
Would a try-with-resources work here? This would eliminate the need to manually close the input stream in all the `if` cases.